File: 백준/Gold/2166. 다각형의 면적/다각형의 면적.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	var n int
	fmt.Fscanf(reader, "%d\n", &n)
	var Xlist = make([]float64, n+1)
	var Ylist = make([]float64, n+1)
	for i := 0; i < n; i++ {
		fmt.Fscanf(reader, "%f %f\n", &Xlist[i], &Ylist[i])
	}
	Xlist[n] = Xlist[0]
	Ylist[n] = Ylist[0]
	var sum float64
	for i := 0; i < len(Ylist)-1; i++ {
		sum += Xlist[i] * Ylist[i+1]
		sum -= Xlist[i+1] * Ylist[i]
	}
	fmt.Printf("%.1f", math.Abs(sum/2))
}

File: 백준/Gold/2166. 다각형의 면적/다각형의 면적.py
import sys 
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
n = int(input())
x_list = []
y_list = []
for _ in range(n):
    x, y =map(int, input().split())
    x_list.append(x)
    y_list.append(y)
x_list.append(x_list[0])
y_list.append(y_list[0])
sum_v = 0
for i in range(len(y_list)-1):
    sum_v+=x_list[i]*y_list[i+1]
    sum_v-=x_list[i+1]*y_list[i]
print("%.1f"% abs(sum_v/2))
